desktop/core/src/desktop/js/ko/components/ko.catalogEntriesList.js

```diff
@@ -330,6 +330,43 @@ class CatalogEntriesList {
       }
     });
 
+    /**
+     * Transforms the data and metadata of a complex response.
+     * This function processes the response based on the column type (map, array, struct)
+     * and appropriately formats the data and metadata for sampling.
+     */
+    self.transformData = function (response, isComplex, dataType) {
+      const encodeToHtml = value => JSON.stringify(value).replace(/"/g, '&quot;');
+      const decodeFromHtml = str => str.replace(/&quot;/g, '"');
+
+      let { data, meta } = response;
+
+      if (isComplex && ['map', 'array', 'struct'].includes(dataType)) {
```

Comment:
Nit: No need for ['map', 'array', 'struct'].includes(dataType) given the if statements below